test/e2e/csc_tests.go

```diff
@@ -0,0 +1,109 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+const (
+	testResourceName string = "test-operators"
+)
+
+func runCSCWithInvalidTargetNamespace(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables
+	f := test.Global
+	// Run tests
+	if err := cscWithInvalidTargetNamespace(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func cscWithInvalidTargetNamespace(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Define variables to be used in test
+	targetNamespace := "nonexisting-namespace"
+	packages := "etcd,federationv2,mongodb,amq-streams,couchbase,descheduler,prometheus,svcat,clusterlogging,dynatrace,metering"
+	registryNamespace := "marketplace_e2e"
+
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Create the operatorsource to download the manifests
+	testOperatorSource := createOperatorSource(testResourceName, namespace, nil, registryNamespace)
+	err = createRuntimeObject(f, ctx, testOperatorSource)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, testResourceName)
+	if err != nil {
+		return err
+	}
+
+	// Delete the existing catalogsourceconfig
+	err = deleteRuntimeObject(f, resultCatalogSourceConfig, testResourceName, namespace)
+	if err != nil {
+		return err
+	}
+
+	// Create a new catalogsourceconfig with a nonexistant targetNamespace
+	testCSC := createCatalogSourceConfig(testResourceName, namespace, targetNamespace, packages)
+	err = createRuntimeObject(f, ctx, testCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with an invalid targetNamespace
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, testResourceName)
+	if err != nil {
+		return err
+	}
+
+	// Check if the catalogsourceconfig phase and message are the expected values
+	expectedPhase := "Configuring"
+	expectedMessage := fmt.Sprintf("namespaces \"%s\" not found", targetNamespace)
+	result := pollUntilTrue(5*time.Second, 1*time.Minute, func() bool {
```

Comment:
Don't we have these timeouts hardcoded elsewhere?

Reply:
There are constants included in the testutils package that define the interval at which we attempt to create a runtime object and the duration we should attempt to do so for. I could update these calls to use those values but the timeout value would decrease which means the tests may fail more. At the very least - the values should be updated to be constants.

Reply:
Yes, please add constants with comments for these.